### ISSUE TITLE
text tool ?

### ISSUE THREAD

**Issue by rgh36167** [mypaint/mypaint#254]
Hi,

is there some way to draw text (other than draw it using a pencil)?

Imho it is either very hard to find or would make a nice addition.

**Comment by achadwick** [comment]
Hi! I assume you're using the development version or a recent alpha? In those, you can use Inkscape to make the text in a (simple) vector layer (layers panel -> context menu -> "New Vector Layer...", or the same command in the Layer menu).

There are no plans to add a text tool in the next release, although a basic system for adding a new vector layer quickly based on text you type in a box could be added later on. That's an approach I favour because it's good for comics: speech balloons.

As far as I know, no other program allows vector layers in OpenRaster, but the Krita devs mentioned adding support for them. Also you will have to distribute or archive any fonts you need alongside the ORA files so that they can be installed and make the text look right: they don't go inside the .ora file right now.

There were some plans for adding text in the OpenRaster spec a while back, but nobody worked on that enough to make it implementable. Getting text layout looking consistent across apps and rendering systems would be a huge nightmare: SVG vector layers were easier. So it goes.

**Comment by speculatrix** [comment]
I installed mypaint for the first time, and looked everywhere to find a way to add text. I find it hard to believe that there isn't even the simplest way to add text to a picture and the developers don't see a need even for the most primitive text addition function possible!
And "just copy some text from another tool" isn't the right answer. I guess I'm going to deinstall mypaint and try something else. It looked so promising too!

**Comment by Starttoaster** [comment]
@speculatrix Also came here in search of a quick way to add text in MyPaint. Kind of shocking that this tool doesn't have a simple text box option. Ever watch that Steve Jobs "documentary" movie with Ashton Kutcher? The scene where Jobs is reaming out his staff, "Somehow we managed  to design a word processor that doesn't have fonts. AND IT CANT HAPPEN AGAIN."

**Comment by jplloyd** [comment]
> @speculatrix Also came here in search of a quick way to add text in MyPaint. Kind of shocking that this tool doesn't have a simple text box option. Ever watch that Steve Jobs "documentary" movie with Ashton Kutcher? The scene where Jobs is reaming out his staff, "Somehow we managed to design a word processor that doesn't have fonts. AND IT CANT HAPPEN AGAIN," comes to mind.

Do you really find it _shocking_ that a piece of software that is defined as a "_drawing and painting_" program does not provide that kind of functionality? I think you could make a case for making a comparison to that movie scene if MyPaint only included a single unconfigurable brush.

At any rate, as stated above, you can use a vector layer to achieve this. The same caveats stated above still apply unfortunately. I'll happily review and merge patches that add functionality to make the process of creating editable text easier, but it's not something I intend to work on myself for the foreseeable future.

**Comment by Starttoaster** [comment]
@jplloyd My apologies as I was actually advertised to MyPaint as sort of a modern day MS Paint competitor. So yes, I was a bit shocked that it seemed to be missing a basic piece of functionality there. My use case was to make simple and quick edits to a few photos for work. Nothing big, mostly just drawing boxes around a few key things and having text off to the side that talks about what is in those boxes. Since MyPaint lacks that functionality (something that MS Paint, Gimp, and Photoshop all have) and is not on the roadmap, I will use a different product. I'm somewhat proficient but maybe a bit slow with Gimp but I guess I will be gaining experience with it shortly.

**Comment by jplloyd** [comment]
@Starttoaster Given your expectations (due to being somewhat misinformed) your reaction is more understandable, no hard feelings.

For the kind of task you have in mind, I would personally probably use Krita or Inkscape, but GIMP should work fine too (in the past their text tool was one of their weaker points, but it's probably better now).

**Comment by Starttoaster** [comment]
@jplloyd Thanks for the tip! Krita was apparently exactly the tool I was looking for; much faster for my simple needs than Gimp.  I appreciate the no hard feelings.

**Comment by zaid-g** [comment]
It's unbelievable that this feature does not exist.

**Comment by Starttoaster** [comment]
Admittedly it would be a nice feature to exist @zaid-g but to be fair, it appears this tool is more centered towards comic strip and electronic based art mediums; and less for "photoshopping" images, for lack of a better term.

**Comment by odysseywestra** [comment]
Yeah, that was what the aim. That way you could pair this up with Inkscape and use it. Plus it allowed us to add svg support to the `ora` files spec as well so tools like Krita can use it too.

**Comment by roberdaniel** [comment]
I want to add my voice to what i think it is a nonsense. Any paint software has a tool for writing. 
I think it is a basic feature.
It does not need to be a full featured tool, only the basics should be fine, ie change font, color and size, maybe a rotation.

**Comment by speculatrix** [comment]
> For the kind of task you have in mind, I would personally probably use Krita 

Yes, I chose krita. Until mypaint gets a basic text tool, I'm sorry to say no matter how good it might be at anything else, it'll be losing a huge "market" share and remain somewhat of a personal project.
